geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/membership/GMSJoinLeave.java

```diff
@@ -349,9 +349,9 @@ public boolean join() throws MemberStartupException {
               && state.possibleCoordinator.equals(this.localAddress)) {
             // if we haven't contacted a member of a cluster maybe this node should
             // become the coordinator.
-            if (state.joinedMembersContacted <= 0 && (now >= locatorGiveUpTime) &&
-                (tries >= minimumRetriesBeforeBecomingCoordinator ||
-                    state.locatorsContacted >= locators.size())) {
+            if (state.joinedMembersContacted <= 0 && (now >= locatorGiveUpTime &&
+                tries >= minimumRetriesBeforeBecomingCoordinator) ||
+                state.locatorsContacted >= locators.size()) {
```

Comment:
It looks like this evaluates to `true` any time we are able to contact all of the locators, even if one of those locators is part of an existing DS. In other words, if `joinedMembersContacted` is greater than zero but `locatorsContacted` is equal to `locators.size()` it tries to become the coordinator. Is that OK/can that even happen? I don't know enough about this code to say for sure.

Reply:
Gah! Good catch! Let me see if I have any more parenthesis lying around here.

Reply:
The new logic in the latest commit makes sense to me.